File: integration-tests/java-selenium/java-selenium-test/src/main/java/com/gu/fronts/integration/test/page/AllFaciaContainersPage.java
package com.gu.fronts.integration.test.page;

import static com.gu.fronts.integration.test.fw.selenium.ByTestAttributeSelector.testAttributeCssSelector;
import static org.openqa.selenium.By.cssSelector;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.gu.fronts.integration.test.fw.selenium.FindByTestAttribute;
import com.gu.fronts.integration.test.page.util.AbstractParentPage;

public class AllFaciaContainersPage extends AbstractParentPage {
    @FindByTestAttribute(using = "all-front-containers")
    private WebElement allFaciaContainers;

    public AllFaciaContainersPage(WebDriver webDriver) {
        super(webDriver);
    }

    @Override
    protected AllFaciaContainersPage isDisplayed() {
        assertExists(allFaciaContainers);
        return this;
    }

    public FaciaContainer containerWithTestAttributeId(String testId) {
        return new FaciaContainer(webDriver, findElementBy(allFaciaContainers,
                cssSelector(testAttributeCssSelector(testId))));
    }
}

File: integration-tests/scala-fw-tests/src/main/scala/com/gu/fronts/integration/test/pages/common/AllFaciaContainersPage.scala
package com.gu.fronts.integration.test.page.common

import org.openqa.selenium.{ WebDriver, WebElement }

import com.gu.fronts.integration.test.pages.common.FrontsParentPage

object AllFaciaContainersPage {
  val IN_PICTURES_CONTAINER_ID = "in-pictures"
}
class AllFaciaContainersPage(implicit var driver: WebDriver) extends FrontsParentPage() {

  private var allFaciaContainers: WebElement = findByTestAttribute("all-front-containers")

  override def isDisplayed() = {
    existsAndDisplayed(allFaciaContainers)
  }

  def containerWithId(testAttributeId: String): FaciaContainer = {
    val faciaContainer = new FaciaContainer(findByTestAttribute(testAttributeId, allFaciaContainers))
    faciaContainer.isDisplayed
    faciaContainer
  }
}